processing/src/main/java/io/druid/segment/filter/BoundFilter.java

```diff
@@ -161,6 +152,25 @@ public boolean supportsBitmapIndex(BitmapIndexSelector selector)
     return selector.getBitmapIndex(boundDimFilter.getDimension()) != null;
   }
 
+  private DruidCompositePredicate getPredicate()
+  {
+    return new DruidCompositePredicate()
+    {
+      @Override
+      public boolean applyLong(long value)
+      {
+        return doesMatch(String.valueOf(value));
```

Comment:
Should do comparisons directly on the longs as part of the work for #2989. But I think for now this approach is okay, since the existing comparators "lexicographic" and "alphaNumeric" are funky and neither of them matches normal numeric comparison.